core/src/main/java/org/apache/iceberg/rest/auth/AuthManager.java

```diff
@@ -0,0 +1,108 @@
+   * credentials, if required, and must be discarded after that.
+   *
+   * <p>This method cannot return null. By default, it returns the catalog session.
+   */
+  default AuthSession initSession(RESTClient initClient, Map<String, String> properties) {
+    return mainSession(initClient, properties);
+  }
+
+  /**
+   * Returns a long-lived session whose lifetime is tied to the owning catalog. This session serves
+   * as the parent session for all other sessions (contextual and table-specific). It is closed when
+   * the owning catalog is closed.
+   *
+   * <p>The provided REST client is a long-lived, shared client; if required, implementors may store
+   * it and reuse it for all subsequent requests to the authorization server, e.g. for renewing or
+   * refreshing credentials. It is not necessary to close it when {@link #close()} is called.
+   *
+   * <p>This method cannot return null.
+   *
+   * <p>It is not required to cache the returned session internally, as the catalog will keep it
+   * alive for the lifetime of the catalog.
+   */
+  AuthSession mainSession(RESTClient sharedClient, Map<String, String> properties);
+
+  /**
+   * Returns a session for a specific context.
+   *
+   * <p>If the context requires a specific {@link AuthSession}, this method should return a new
+   * {@link AuthSession} instance, otherwise it should return the parent session.
+   *
+   * <p>This method cannot return null. By default, it returns the parent session.
+   *
+   * <p>Implementors should cache contextual sessions internally, as the catalog will not cache
+   * them. Also, the owning catalog never closes contextual sessions; implementations should manage
+   * their lifecycle themselves and close them when they are no longer needed.
+   */
+  default AuthSession contextualSession(SessionCatalog.SessionContext context, AuthSession parent) {
+    return parent;
+  }
+
+  /**
+   * Returns a new session targeting a specific table or view. The properties are the ones returned
+   * by the table/view endpoint.
+   *
+   * <p>If the table or view requires a specific {@link AuthSession}, this method should return a
+   * new {@link AuthSession} instance, otherwise it should return the parent session.
+   *
+   * <p>This method cannot return null. By default, it returns the parent session.
+   *
+   * <p>Implementors should cache table sessions internally, as the catalog will not cache them.
+   * Also, the owning catalog never closes table sessions; implementations should manage their
+   * lifecycle themselves and close them when they are no longer needed.
+   */
+  default AuthSession tableSession(
+      TableIdentifier table, Map<String, String> properties, AuthSession parent) {
+    return parent;
+  }
+
+  /**
+   * Closes the manager and releases any resources.
+   *
+   * <p>This method is called when the owning catalog is closed.
+   */
+  @Override
+  default void close() {
```

Comment:
is there any value in having a default impl that doesn't do anything? I feel like this should be just left to the actual auth manager implementations